### PR TITLE
Misc test fixes

### PR DESCRIPTION
**Pull request by rustyrussell**
This contains some of the simpler fixes which crept into the ongoing permconnect (#1322) which is taking far too long.  Let's make sure these pass CI and get reviewed, while I work on that one.

### CODE REVIEW

File: lightningd/peer_control.c [hunk 1]
```diff
@@ -917,10 +933,8 @@ static void activate_peer(struct peer *peer)
 
 	/* We can only have one active channel: reconnect if not already. */
 	channel = peer_active_channel(peer);
-	if (channel && !channel->owner) {
-		msg = towire_gossipctl_reach_peer(peer, &peer->id);
-		subd_send_msg(peer->ld->gossip, take(msg));
-	}
+	if (channel && !channel->owner)
```
Comment (ZmnSCPxj) [inline lightningd/peer_control.c]: Good catch!

**Comment by ZmnSCPxj** [comment]
ACK 41a8f7c25d3022c59b9d5b73183197d760bb7e2a

**Comment by rustyrussell** [comment]
Slight modification, I missed another test which may reconnect (if we're slow enough); I changed connect() to suppress reconnects which turns out to be what all the callers want.

**Comment by ZmnSCPxj** [comment]
ACK a2c97aed961f221cf6709287138699e433d21113

**Pull request merged**